File: frontend/src/models/EngineerUserAbilityReport.ts
export default interface EngineerUserAbilityReport {
  id: number;
  engineerUserId: number;
  projectScore: number;
  repositoryScore: number;
  commitScore: number;
  pullreqScore: number;
  issueScore: number;
  speedScore: number;
  createdAt: Date;
}

export function calcEngineerScore(ability: EngineerUserAbilityReport) {
  return calcDetectabilityScore(ability)
    + calcSolvingScore(ability)
    + calcSpeedScore(ability);
}

export function calcDetectabilityScore(ability: EngineerUserAbilityReport) {
  return ability.issueScore;
}

export function calcSolvingScore(ability: EngineerUserAbilityReport) {
  console.log(ability)
  return ability.projectScore
    + ability.repositoryScore
    + ability.commitScore
    + ability.pullreqScore;
}

export function calcSpeedScore(ability: EngineerUserAbilityReport) {
  return ability.speedScore;
}

File: backend/model/engineer_user_ability_report.go
package model

import "time"

type EngineerUserAbilityReport struct {
	ID              uint64    `db:"id" json:"id"`
	EngineerUserID  uint64    `db:"engineer_users_id" json:"enginerUserId"`
	ProjectScore    uint64    `db:"project_score" json:"projectScore"`
	RepositoryScore uint64    `db:"repository_score" json:"repositoryScore"`
	CommitScore     uint64    `db:"commit_score" json:"commitScore"`
	PullreqScore    uint64    `db:"pullreq_score" json:"pullreqScore"`
	IssueScore      uint64    `db:"issue_score" json:"issueScore"`
	SpeedScore      uint64    `db:"speed_score" json:"speedScore"`
	CreatedAt       time.Time `db:"created_at" json:"createdAt"`
}
